TACC-Cloud/hazmapper
# task/WG-236: Websocket notifications

## Overview: ##

## PR Status: ##


## Related Jira tickets: ##

* [WG-236](https://tacc-main.atlassian.net/browse/WG-236)

## Summary of Changes: ##

- Added WebSocket Status (Connected or Disconnected so user/devs can see that websocket is connected and working. This is a temp UI feature and will be removed before merge)
- Added a Trigger Notification button (temp and will be removed upon PR reviews and before merge) so that devs can see Toast notifications pop up in the client.
- There's a "Trigger Notification" button on the Main Menu, that when clicked shows a general info notification to show it's working.
- Also added two Asset buttons in the Assets Panel. One called Asset Success and one called Asset Failure. This is to simulate successfully adding an asset and/or an asset failing to be added. Clicking both buttons triggers a notification on the Map. (These are temp and will be removed. They're just for dev and testing purposes to trigger events, while we wait to complete the Asset Panel and Adding Asset features).
- Made a socketUtils file that contains all the socket utilities/functions for easier use across the application.

## Testing Steps: ##
1. Run with 'task/WG-236-websocket-notifications' backend branch from the geoapi repo [Backend](https://github.com/TACC-Cloud/geoapi/pull/185)
2. Click the "Trigger Notification" button on the Main Menu of React Hazmapper
3. Should see a toast notification pop up in the right corner. Compare with first video below
4. Create a new map project or go to an existing map project
5. Click the "Assets" tab in the Panel
6. Click the "Asset Success" and "Asset Failure" buttons in the Asset Panel to see the toast notifications that will be related to importing assets (once that feature is added).
7. Compare with second video below

## UI Photos:

https://github.com/TACC-Cloud/hazmapper/assets/50084480/5924a110-1c2b-4a7f-a931-e396d6e242b0

https://github.com/TACC-Cloud/hazmapper/assets/50084480/821cc335-0de9-49c4-9254-be1b698685d9



## Notes: ##
